src/react/webgl-renderer.js

```diff
@@ -131,9 +131,52 @@ export default class WebGLRenderer extends React.Component {
     }
   }
 
+  // Gets current size of canvas in css (logical/window) coordinates
+  _getCSSSize(canvas) {
+    return {
+      width: canvas.clientWidth,
+      height: canvas.clientHeight
+    };
+  }
+
+  // Calculate the drawing buffer size that would cover current canvas size and device pixel ratio
+  // Intention is that every pixel in the drawing buffer will have a 1-to-1 mapping with
+  // actual device pixels in the hardware framebuffer, allowing us to render at the full
+  // resolution of the device.
+  _calculateDrawingBufferSize(canvas, {useDevicePixelRatio = true}) {
+    const cssToDevicePixels = useDevicePixelRatio ? window.devicePixelRatio || 1 : 1;
+
+    // Lookup the size the browser is displaying the canvas in CSS pixels
+    // and compute a size needed to make our drawingbuffer match it in
+    // device pixels.
+    const cssSize = this._getCSSSize(canvas);
+    return {
+      width: Math.floor(cssSize.width * cssToDevicePixels),
+      height: Math.floor(cssSize.height * cssToDevicePixels),
+      devicePixelRatio: cssToDevicePixels
+    };
+  }
+
+  // Resizes canvas width and height to match with device drawing buffer
+  _resizeDrawingBuffer(canvas, {useDevicePixelRatio = true}) {
+    // Resize the render buffer of the canvas to match canvas client size
+    // multiplying with dpr (Optionally can be turned off)
+    const newBufferSize = this._calculateDrawingBufferSize(canvas, {useDevicePixelRatio});
+    // Only update if the canvas size has not changed
+    if (newBufferSize.width !== canvas.width || newBufferSize.height !== canvas.height) {
+      // Make the canvas render buffer the same size as
+      canvas.width = newBufferSize.width;
+      canvas.height = newBufferSize.height;
+      // Always reset CSS size after setting drawing buffer size
+      // canvas.style.width = `${cssSize.width}px`;
+      // canvas.style.height = `${cssSize.height}px`;
+    }
+  }
+
   // Updates WebGL viewport to latest props
   // for clean logging, only calls gl.viewport if props have changed
   _updateGLViewport() {
+    /*
     let {viewport: {x, y, width: w, height: h}} = this.props;
```

Comment:
The size in props is not multiplied by `devicePixelRatio`. It is the `width` and `height` that users pass to the `DeckGL` component.